### PR TITLE
External tests maintenance

### PR DESCRIPTION
**Pull request by nikola-matic**
- [x] Added `reports/` directory to `.gitignore`, generated by running external tests.
- [x] Changed [hardhat/2115](https://github.com/NomicFoundation/hardhat/issues/2115) and [hardhat/2453](https://github.com/NomicFoundation/hardhat/issues/2453) to point to [hardhat/3365](https://github.com/NomicFoundation/hardhat/issues/3365), as the former issues were closed in lieu of 3365, which is meant to encompass all such issues. I've also ping @fvictorio to see if [hardhat/2051](https://github.com/NomicFoundation/hardhat/issues/2051) falls under the same umbrella.
- [x] Removed peg from `ens.sh` as the underlying [issue](https://github.com/Uniswap/v3-periphery/issues/313) was fixed.
- [x] Removed hardhat peg from `gnosis.sh` as the underlying [issue](https://github.com/NomicFoundation/hardhat/issues/2709) was fixed. This fixes a part of https://github.com/ethereum/solidity/issues/13051, but unfortunately the rest of the issues have not been solved, and the issue can thus not be closed.

### CODE REVIEW

File: test/externalTests/ens.sh [hunk 1]
```diff
@@ -69,9 +69,9 @@ function ens_test
     neutralize_packaged_contracts
 
     # In some cases Hardhat does not detect revert reasons properly via IR.
-    # TODO: Remove this when https://github.com/NomicFoundation/hardhat/issues/2115 gets fixed.
+    # TODO: Remove this when https://github.com/NomicFoundation/hardhat/issues/3365 gets fixed.
     sed -i "s|it\(('Does not allow wrapping a name you do not own',\)|it.skip\1|g" test/wrapper/NameWrapper.js
-    # TODO: Remove this when https://github.com/NomicFoundation/hardhat/issues/2453 gets fixed.
+    # TODO: Remove this when https://github.com/NomicFoundation/hardhat/issues/3365 gets fixed.
```
Comment (cameel) [inline test/externalTests/ens.sh]: We could just merge with the section above.
```suggestion
```

**Review by cameel — DISMISSED** [review]

**Comment by cameel** [comment]
By the way, this is a situation where separate commits would be perfectly appropriate. Each one of the points in the description is pretty much an independent change.

**Comment by nikola-matic** [comment]
> By the way, this is a situation where separate commits would be perfectly appropriate. Each one of the points in the description is pretty much an independent change.

Separate commits or PRs? I theory I agree, but having 3 commits seems like overkill to me, and we're gonna squash them in the anyway, no?

File: test/externalTests/ens.sh (hunk 1 above)
Comment (nikola-matic) [reply test/externalTests/ens.sh]: Good point, didn't even notice. `%s/XXXX/3365/g` goes brrr.

**Review by cameel — APPROVED** [review]

**Comment by cameel** [comment]
> Separate commits or PRs? I theory I agree, but having 3 commits seems like overkill to me, and we're gonna squash them in the anyway, no?

Commits. And no, we don't have to squash them. We only really squash review fixes into the original commits. Or we squash everything when the commits are so messy that they make the history harder to understand than just having a single commit. But overall all that we require is clean history, going straight to the point without any of the dead ends you went through when working on a solution, that is easy to understand when you look at it in git. Having each logical change in a separate, atomic commit with a clear message is fine and actually recommended.

**Pull request merged**